power_grid_model_c/power_grid_model/include/power_grid_model/calculation_parameters.hpp

```diff
@@ -159,6 +159,11 @@ template <bool sym> struct MathModelParam {
     ComplexTensorVector<sym> source_param;
 };
 
+template <bool sym> struct MathModelParamIncrement : public MathModelParam<sym> {
```

Comment:
I am guessing out of laziness

Reply:
Actually it's neither:
`
auto original = MathModelParam{
  .branch_params = {branch_param_0, branch_param_1 /*non-zero*/, branch_param_2 /*non-zero*/}
};

auto increment = MathModelParamIncrement{
  .params = {branch_param_0, branch_param_1, branch_param_2},
  .to_change = {1, 2}
};
`
The original sparse parameters are kept to avoid back and forth index conversion

Reply:
sooooo does the increment version contain a copy of the original components that are not muted then?

Reply:
If by 'original components` you mean original `MathModelParam`, yes; in the increment mode, the input `MathModelParam` is assumed to be deltas already. I.e.,
```python
auto original_input_params = MathModelParam{
    .params = { branch_param_delta_0, branch_param_delta_1, branch_param_delta_2 }
};
```
and the increment version:
```python
auto original_input_params = MathModelParam{
    .params = { branch_param_delta_0, branch_param_delta_1, branch_param_delta_2 }
    .to_change = { 1, 2 }
};
```